src/main/java/com/powsybl/sld/server/PositionDiagramLabelProvider.java

```diff
@@ -0,0 +1,148 @@
+        checkConnectableInVoltageLevel(singleOrder, voltageLevel, injection, throwException);
+        return singleOrder;
+    }
+
+    static Integer getBranchOrder(ConnectablePosition<?> position, VoltageLevel voltageLevel, Branch<?> branch, boolean throwException) {
+        Integer order;
+        if (branch.getTerminal1().getVoltageLevel() == voltageLevel) {
+            order = position.getFeeder1().getOrder().orElse(null);
+        } else if (branch.getTerminal2().getVoltageLevel() == voltageLevel) {
+            order = position.getFeeder2().getOrder().orElse(null);
+        } else {
+            throw new PowsyblException(String.format("Given voltageLevel %s not found in terminal 1 and terminal 2 of branch", voltageLevel.getId()));
+        }
+        checkConnectableInVoltageLevel(order, voltageLevel, branch, throwException);
+        return order;
+    }
+
+    static Integer get3wtOrder(ConnectablePosition<?> position, VoltageLevel voltageLevel, ThreeWindingsTransformer twt, boolean throwException) {
+        Integer order;
+        if (twt.getLeg1().getTerminal().getVoltageLevel() == voltageLevel) {
+            order = position.getFeeder1().getOrder().orElse(null);
+        } else if (twt.getLeg2().getTerminal().getVoltageLevel() == voltageLevel) {
+            order = position.getFeeder2().getOrder().orElse(null);
+        } else if (twt.getLeg3().getTerminal().getVoltageLevel() == voltageLevel) {
+            order = position.getFeeder3().getOrder().orElse(null);
+        } else {
+            throw new PowsyblException(String.format("Given voltageLevel %s not found in leg 1, leg 2 and leg 3 of ThreeWindingsTransformer", voltageLevel.getId()));
+        }
+        checkConnectableInVoltageLevel(order, voltageLevel, twt, throwException);
+        return order;
+    }
+
+    static Integer getOrderPositions(ConnectablePosition<?> position, VoltageLevel voltageLevel, Identifiable<?> identifiable, boolean throwException) {
+        if (identifiable instanceof Injection) {
+            return getInjectionOrder(position, voltageLevel, (Injection<?>) identifiable, throwException);
+        } else if (identifiable instanceof Branch) {
+            return getBranchOrder(position, voltageLevel, (Branch<?>) identifiable, throwException);
+        } else if (identifiable instanceof ThreeWindingsTransformer) {
+            return get3wtOrder(position, voltageLevel, (ThreeWindingsTransformer) identifiable, throwException);
+        } else {
+            LOGGER.error("Given connectable not supported: {}", identifiable.getClass().getName());
+            if (throwException) {
+                throw new PowsyblException(String.format("Given connectable %s not supported: ", identifiable.getClass().getName()));
+            }
+        }
+        return null;
+    }
+
+    static void checkConnectableInVoltageLevel(Integer order, VoltageLevel voltageLevel, Connectable<?> connectable, boolean throwException) {
```

Comment:
Do you really need this? It does not seem to do what it is named.

Reply:
i think we don't need it for now.